### PR TITLE
Package samples code in PyPi package

### PR DESCRIPTION
**Pull request by eccles**
Problem:
Common usage requires the samples code be installed via a wheel.

Solution:
Added all hooks for creating and publishing wheel

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

**Review by robinbryce — COMMENTED** [review]
I like entrypoints a lot. Great change in general.

The reference for entry points and the first two credible looking guides I can find on the use of entrypoins use "package:main" most of the time, and occasionally "packge:specific_function"

And I find

`archivist_samples_door_entry = archivist_samples.door_entry.entry:entry` confusing

I think we should stick to the established convention of using 'main' as the name for the entry point target function OR a function name specific to what the ep does.

Unless you can point me at some precedent that illustrates the what & the why of this choice ?

1. https://setuptools.readthedocs.io/en/latest/userguide/entry_point.html
2. https://amir.rachum.com/blog/2017/07/28/python-entry-points/
3. https://chriswarrick.com/blog/2014/09/15/python-apps-the-right-way-entry_points-and-scripts/

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    author="Jitsuin Inc.",
+    author_email="support@jitsuin.com",
+    description="Jitsuin Archivist Examples",
+    long_description=DESC,
+    long_description_content_type="text/markdown",
+    url=REPO_URL,
+    packages=find_packages(exclude=( "examples", "unittests", )),
+    platforms=['any'],
+    classifiers=[
+        'Development Status :: 3 - Alpha', #pre-delivery
```
Comment (robinbryce) [inline setup.py:30]: Is there a SPDX identifier we can use here as well or instead of ?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    version_config={
+        "template": "{tag}",
+        "dev_template": "{tag}.post{ccount}+git.{sha}",
+        "dirty_template": "{tag}.post{ccount}+git.{sha}.dirty",
+        "version_callback": None,
+        "version_file": None,
+        "count_commits_from_version_file": False,
+    },
+    setup_requires=['setuptools-git-versioning'],
+    python_requires='>=3.6',
```
Comment (robinbryce) [inline setup.py:48]: If the tests introduce dependencies I think there is also a tests_requires we can use here

File: setup.py (hunk 1 above)
Comment (eccles) [reply setup.py:30]: The standard pypi license iclassifier is from here: https://gist.github.com/nazrulworld/3800c84e28dc464b2b30cec8bc1287fc

                    License :: OSI Approved :: MIT License

File: setup.py (hunk 2 above)
Comment (eccles) [reply setup.py:48]: there are no tests in this repo...

**Review by robinbryce — APPROVED** [review]
Great step forward. Thanks!

**Pull request merged**